cli/kaos_cli/utils/validators.py

```diff
@@ -8,7 +8,77 @@
 import textdistance
 from kaos_cli.exceptions.exceptions import MissingArgumentError
 
-from ..constants import KAOS_STATE_DIR, DOCKER, MINIKUBE, KAOS_TF_PATH
+from ..constants import KAOS_STATE_DIR, DOCKER, MINIKUBE, KAOS_TF_PATH, TF_STATE
+
+
+class EnvironmentState:
+    """
+    This class is used to validate existence of already built environmnets
+    """
+    def __init__(self):
+        self.cloud = None
+        self.env = None
+        self.build_dir = None
+        self.tf_state_path = None
+        self.if_build_dir_exists = None
+        self.if_tfstate_exists = None
+
+    @classmethod
+    def validate_build_states(cls, cloud, env) -> "EnvironmentState":
+        env_state = cls()
+        env_state.cloud = cloud
+        env_state.env = env
+        env_dir = f"{env_state.cloud}/{env_state.env}" if env_state.cloud not in [DOCKER, MINIKUBE] \
+            else f"{env_state.cloud}"
+        build_dir = os.path.join(KAOS_TF_PATH, env_dir)
+        tf_state_path = os.path.join(KAOS_TF_PATH, env_dir, f"{TF_STATE}")
+        env_state.if_build_dir_exists = os.path.exists(build_dir)
+        env_state.if_tfstate_exists = os.path.exists(tf_state_path)
+        env_state.build_dir = build_dir
+        env_state.tf_state_path = tf_state_path
+        return env_state
+
+    def display_warning(self) -> "EnvironmentState":
+        """
+        Ensure existence of kaos backend dir (tf_path) and throw appropriate warnings if it does not exist
+        """
+        if not self.if_tfstate_exists:
+            if self.env:
+                    click.echo("{} - {} [{}] backend in {} has not been deployed!".format(
+                        click.style("Warning", bold=True, fg='yellow'),
+                        click.style('kaos', bold=True),
+                        click.style(self.env, bold=True, fg='blue'),
+                        click.style(self.cloud, bold=True, fg='red')
+                    ))
+            elif not self.env:
+                click.echo("{} - {} backend in {} has not been deployed!".format(
+                    click.style("Warning", bold=True, fg='yellow'),
+                    click.style('kaos', bold=True),
+                    click.style(self.cloud, bold=True, fg='red')
+                ))
+            sys.exit(1)
+        else:
+            pass
+
+    def set_build_env(self) -> "EnvironmentState":
+        """
+        Simple method to ensure the right env is set for local v cloud kaos build
+        """
+
+        try:
+            if self.env and self.cloud in [DOCKER, MINIKUBE]:
+                click.echo("{} - {} (-e/--env) is not applicable for {} deployment".format(
+                    click.style("Warning", bold=True, fg='yellow'),
+                    click.style("ENV", bold=True),
+                    click.style(self.cloud, bold=True, fg='red')))
+                self.env = None
+            elif self.cloud in [DOCKER, MINIKUBE]:
+                self.env = None
```

Comment:
I would simplify this block into:

```
if self.env in [AWS, GCP, AZURE]:
    # do some cloud env stuff
elif self.env in [DOCKER, MINICUBE]:
    # None-ify it
else:
    # either self.env is None or something else
    # None-ify it?

Reply:
In this case we are actually validating the presence/absence of `self.env` variable which could have values in ['None', 'dev'. 'stage', 'prod'] based on the values of `self.cloud` which could have values in ['None', 'GCP'. 'AWS', 'DOCKER']. So, I feel the block that we currently have is the simplest way to resolve and set `env` variable based on the associated `cloud` value (None in case `self.cloud` in [DOCKER, MINICUBE] and defaulting `prod` value in case of cloud installation and `self.env`is not set)